src/arch/riscv/pagetable_walker.cc

```
@@ -89,10 +89,304 @@ Walker::WalkerStats::WalkerStats(statistics::Group *parent)
+        } else {
+            remaining.push_back(mq_entry);
+        }
+    }
+
+    ptwMissQueue.swap(remaining);
+
+    while (!matched.empty()) {
+        MissQueueEntry mq_entry = matched.front();
+        matched.pop_front();
+        stats.ptwMissQueueHintRetries++;
+        bool resolved = tlb->retryTimingPtwMiss(mq_entry.tc,
```

Cppcheck (2.21.0)

[error] Shifting by a negative value is undefined behaviour

(shiftNegative)